sqlglot/optimizer/qualify_columns.py

```diff
@@ -258,12 +260,18 @@ def _qualify_columns(scope, resolver):
             column.set("table", column_table)
 
 
-def _expand_stars(scope, resolver):
+def _expand_stars(scope, resolver, using_column_tables):
     """Expand stars to lists of column selections"""
 
     new_selections = []
     except_columns = {}
     replace_columns = {}
+    coalesced_columns = set()
+
+    # Create a mapping of automatically joined column names to sets of
+    # the corresponding tables, to make the lookup more efficient
+    ordered_column_tables = using_column_tables
```

Comment:
Perhaps we should use some kind of ordered set when we originally construct this object? I.e. a dict with `None` values.

Reply:
We could, yeah. I think I like this idea as it'll simplify the code here.

Reply:
On the other hand, maybe the current approach is a bit cleaner.. Doing `tables[table] = None` in `_expand_using` might look a bit confusing.

Reply:
Done